apps/shell/app.js

```diff
@@ -23,6 +25,161 @@ if (fs.existsSync('.env')) {
   dotenv.config({path: '.env'});
 }
 
+//Start Terminal Color Scheme Implementation
+//declared constants for directory paths
+const xdg_config_dir = (process.env["XDG_CONFIG_HOME"] || path.join(os.homedir(), ".config"));
+const ood_app_config_root = (process.env["OOD_APP_CONFIG_ROOT"] || '/etc/ood/config/ondemand/apps/shell');
+const userDir = path.join(xdg_config_dir, "ondemand", "apps", "shell", "themes");
+const systemDir = path.join(ood_app_config_root, "themes");
+
+//declared global variables for color schemes
+var schemeObjects;
+
+checkDirSync(userDir);
+checkDirSync(systemDir);
+
+if (process.env["SYSTEM_SCHEMES"] === true) {
+    var user = getSchemeObjects(userDir);
+    var system = getSchemeObjects(systemDir);
+
+   schemeObjects = {...getSchemeObjects(userDir), ...getSchemeObjects(systemDir)};
+
+} else {
+   schemeObjects = {...getSchemeObjects(userDir)};
+}
+
+// helper functions
+function checkDirSync(dir) {  
+  try {
+    fs.statSync(dir);
+  } catch(e) {
+    if (dir === systemDir) {
+      process.env["SYSTEM_SCHEMES"] = false;
+    } else if (dir === userDir) {
+      fs.mkdirSync(dir);
+    }
+  }
+}
+
+function getSchemeObjects(dir) {
+  var schemes = {};
+
+  fs.readdirSync(dir).forEach(function(file) {
+    fileInfo = path.parse(file);
+    schemes[fileInfo.name] = {name: fileInfo.name, file: fileInfo.base, ext: fileInfo.ext, dir: dir}
+  });
+  return schemes;
+}
+
+function getSchemeFilesArray() {
+
+    return Object.keys(schemeObjects).map(i => schemeObjects[i])
+}
+
+
+function getSchemeFileObject(base) {
+    var userDir = path.join(xdg_config_dir, "apps", "shell", "themes");
+    fs.readdirSync(userDir).forEach(file => {
+        if (file === base) {
+            return path.parse(file);
+        }
+
+    })
+}
+
+function rgbToHexMath (num) { 
```

Comment:
Add some documentation above all of these functions following https://jsdoc.app/ format